games-emulation/fbalpha_libretro/fbalpha_libretro-0.2.97.42~git.recipe

```diff
@@ -0,0 +1,50 @@
+SUMMARY="A port of Final Burn Alpha to the libretro architecture"
+DESCRIPTION="Final Burn Alpha is a multi-systems Arcade Emulator most popular \
+for emulating Neo-Geo, Capcom, Konami, and Cave games. It is developed by the \
+Final Burn team and originated from FinalBurn by Dave."
+HOMEPAGE="https://github.com/libretro/fbalpha"
+COPYRIGHT="2000-2018 the Final Burn team, the libretro team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+srcGitRev="099c24b166f08b671f703051de1daffb2e6d5eb7"
+SOURCE_URI="https://github.com/libretro/fbalpha/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="2ac4a4cf35aaf3380195b4804d9574be33ae52f4f95b7bd3ab8be450de5c8673"
+SOURCE_FILENAME="fbalpha-$portVersion~git.tar.gz"
```

Comment:
```diff
-SOURCE_FILENAME="fbalpha-$portVersion~git.tar.gz"
+SOURCE_FILENAME="fbalpha-$portVersion-$srcGitRev.tar.gz"
```
(The `~git` is already in `$portVersion` so we don't need to add another one here.)

Reply:
Ok, I'm still trying to understand what the policy is here :D I guess we can use only ~git and update the package revision if we ever refresh it with the same version but a more recent git revision.

Reply:
We use `~git.recipe` (instead of `.recipe`) whenever we have a recipe which uses a srcGitRev to point to some git revision that is not available in a stable release.

When you run `haikuporter fbalpla_libretro-0.2.97.42~git` (or just  `haikuporter fbalpla_libretro` if that's the most recent version), haikuporter defines several shell variables, among which you have portVersion which is defined to the string `0.2.97.42~git`. This string is the name of the recipe file without the `.recipe` suffix and without every letter from the beginning up to the first dash. (It drops the `fbalpla_libretro-`.)
That's why the `~git` becomes part of the `$portVersion`.
Here, with a `SOURCE_FILENAME="fbalpha-$portVersion-$srcGitRev.tar.gz"` the tarball will be called `fbalpha-0.2.97.42~git-099c24b166f08b671f703051de1daffb2e6d5eb7.tar.gz`.

Reply:
> I guess we can use only ~git and update the package revision if we ever refresh it with the same version but a more recent git revision.

Exactly.

Reply:
Hmmm in this case it's a bit different then, here 0.2.97.42 is the latest stable release, but the source code on Git is more recent. Does naming it 0.2.97.42~git still work?

Reply:
There is currently no tarball for v0.2.97.42 although we can see this commit: [Bump to v0.2.97.42](https://github.com/libretro/fbalpha/commit/ed1e7061cb4500ac57b3f7aa4245de66b28c26e9).

This means we should either :
* drop the `~git` (in the name of the recipe) but keep the srcGitRev and increment `REVISION` everytime we update srcGitRev, until 0.2.97.43 becomes available, or
* rename the recipe to `fbalpha_libretro-0.2.97.43~git.recipe`.

> Does naming it 0.2.97.42~git still work?

As you suspected, it's not recommended to keep the `~git` (after `0.2.97.42` in the name of the recipe) if that version is already officially out.
It would work, but if a stable tarball for v0.2.97.42 is made available with contents older than the git revision we have in srcGitRev then it would be misleading, as we would be using something newer than the official realease.

Recipe name | Pros | Cons
-- | -- | --
fbalpha_libretro-0.2.97.42~git.recipe | - | _"Hey, HaikuPorts thinks `v0.2.97.42` is not yet available!!!"_
fbalpha_libretro-0.2.97.42.recipe | _"HaikuPorts knows `v0.2.97.42` is available."_ | _"Is this really `v0.2.97.42` or is it `v0.2.97.43~git`?"_
fbalpha_libretro-0.2.97.43~git.recipe | More appropriate. | _"Why does the app reports `v0.2.97.43` if that release is not out?"_

Well, we could choose `fbalpha_libretro-0.2.97.43~git.recipe` and change `BUILD()` a little to have the app report `v0.2.97.43~git`(instead of `v0.2.97.43`):
```diff
-	sed -e "s/@DISPLAY_VERSION@/v${portVersion%~.*}/" \
+	sed -e "s/@DISPLAY_VERSION@/v$portVersion/" \
 		$portDir/additional-files/fbalpha_libretro.info.in \
 		> fbalpha_libretro.info
```
What do you think?